### PR TITLE
adding atsas-pipelines recipe

### PR DESCRIPTION
**Pull request by kbeeperez**
implements: https://github.com/nsls-ii-forge/atsas-pipelines-feedstock

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/atsas-pipelines/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  commands:
+    - pytest --pyargs atsas_pipelines.tests
+
+about:
+  home: https://github.com/NSLS-II-LIX/atsas-pipelines
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE
+  summary: ATSAS pipelines for the NSLS-II LIX beamline
+  doc_url: https://mrakitin.github.io/atsas-pipelines
```
Comment (mrakitin) [inline recipes/atsas-pipelines/meta.yaml:48]:
```suggestion
  doc_url: https://nsls-ii-lix.github.io/atsas-pipelines
```

**Review by mrakitin — COMMENTED** [review]
I spotted the outdated doc link. Please accept the new suggested URL.

**Comment by mrakitin** [comment]
@conda-forge/staged-recipes, I agree to be a maintainer of this package (that I developed with my NSLS-II colleagues).

The CI on Windows fails with the following error:
```
WARNING: failed to get_linkages, codefile_type('C:\Miniconda\conda-bld\atsas-pipelines_1624574852494\_h_env\Scripts\dask-slurm-cluster.exe')=EXEfile
   INFO: sysroot: 'C:/Windows/' files: '['winhlp32.exe', 'win.ini', 'twain_32/wiatwain.ds', 'twain_32.dll']'
   INFO (atsas-pipelines): Interpreted package 'atsas-pipelines' is interpreted by 'python'
[noarch_python] Noarch package contains binary script: dask-slurm-cluster.exe
```

We do have an entry point in the code: https://github.com/NSLS-II-LIX/atsas-pipelines/blob/fbb5cc1be7fe38f1e89cc1b6ca0579e0bbe1c19c/setup.py#L50. However, I am not sure why it can't be `noarch` with that. Could you please clarify? Should we add an entry point explicitly here in the `build` section, like it's done in https://github.com/nsls-ii-forge/bluesky-browser-feedstock/blob/53703f9d6308b549addc54235a02f396471b689e/recipe/meta.yaml#L18-L20?

File: recipes/atsas-pipelines/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (mrakitin) [inline recipes/atsas-pipelines/meta.yaml:16]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps
```

File: recipes/atsas-pipelines/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+    - dask-jobqueue
+    - distributed
+    - matplotlib
```
Comment (dopplershift) [inline recipes/atsas-pipelines/meta.yaml:26]: Do you need a PyQt GUI? If not, better to depend on `matplotlib-base`.

Comment (mrakitin) [reply recipes/atsas-pipelines/meta.yaml:26]: No, no need for the Qt part for this project. Thanks for the pointer, @dopplershift!

Comment (mrakitin) [inline recipes/atsas-pipelines/meta.yaml:26]:
```suggestion
    - matplotlib-base
```

File: recipes/atsas-pipelines/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
```
Comment (dopplershift) [inline recipes/atsas-pipelines/meta.yaml:17]:
```suggestion
entry_points:
  - dask-slurm-cluster = atsas_pipelines.cli:run_cluster

```

File: recipes/atsas-pipelines/meta.yaml (hunk 3 above)
Comment (mrakitin) [reply recipes/atsas-pipelines/meta.yaml:26]: Actually, I've checked, and I don't see the `matplotlib` imports anywhere. I don't remember why I added it as a dependency, probably was used for some checks at the very beginning of the project. It's OK to remove it from here, and I'll update the project to exclude it from the requirements.

File: recipes/atsas-pipelines/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv<span class="x x-first x-last"> --no-deps</span>
+entry_points:
+  - dask-slurm-cluster = atsas_pipelines.cli:run_cluster
```
Comment (mrakitin) [inline recipes/atsas-pipelines/meta.yaml:18]:
```suggestion
  entry_points:
    - dask-slurm-cluster = atsas_pipelines.cli:run_cluster
```

File: recipes/atsas-pipelines/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv<span class="x x-first x-last"> --no-deps</span>
```
Comment (mrakitin) [inline recipes/atsas-pipelines/meta.yaml:16]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps
```

**Comment by mrakitin** [comment]
All good now! Even the Windows build passes. Thank you, @kbeeperez and @dopplershift!

**Review by dopplershift — APPROVED** [review]

**Pull request merged**